contracts/0.8.25/vaults/FlashMinter.sol

```diff
@@ -0,0 +1,101 @@
+
+    constructor(IStETH _stETH) VaultHub(_stETH) {}
+
+    /**
+     * @notice Register an address as a recipient for flash minting
+     * @param _recipient The address of the recipient to register
+     */
+    function registerFlashMintRecipient(address _recipient) external onlyRole(FLASH_MINT_RECIPIENT_REGISTRY_ROLE) {
+        if (_recipient == address(0)) revert ZeroArgument("_recipient");
+        if (isRegisteredFlashMintRecipient[_recipient]) revert FlashMintRecipientAlreadyRegistered(_recipient);
+
+        isRegisteredFlashMintRecipient[_recipient] = true;
+
+        emit FlashMintRecipientAdded(msg.sender, _recipient);
+    }
+
+    /**
+     * @notice Unregister an address as a recipient for flash minting
+     * @param _recipient The address of the recipient to unregister
+     */
+    function unregisterFlashMintRecipient(address _recipient) external onlyRole(FLASH_MINT_RECIPIENT_REGISTRY_ROLE) {
+        if (_recipient == address(0)) revert ZeroArgument("_recipient");
+        if (!isRegisteredFlashMintRecipient[_recipient]) revert FlashMintRecipientNotRegistered(_recipient);
+
+        delete isRegisteredFlashMintRecipient[_recipient];
+
+        emit FlashMintRecipientRemoved(msg.sender, _recipient);
+    }
+
+    function flashMintShares(
+        address _vault,
+        address _recipient,
+        uint256 _amountOfShares,
+        bytes calldata _data
+    ) external whenResumed {
+        // basic checks & auth
+        if (_vault == address(0)) revert ZeroArgument("_vault");
+        if (_amountOfShares == 0) revert ZeroArgument("_amountOfShares");
+        // allow only registered recipients or the vault owner itself for flash minting
+        if (!isRegisteredFlashMintRecipient[_recipient] && _recipient != OwnableUpgradeable(_vault).owner())
+            revert FlashMintRecipientInvalid(_recipient);
+
+        _vaultAuth(_vault, "flashMint");
+
+        // cannot result in shares exceeding the share limit
+        VaultSocket storage socket = _connectedSocket(_vault);
+
+        uint256 vaultSharesAfterMint = socket.sharesMinted + _amountOfShares;
+        uint256 shareLimit = socket.shareLimit;
+        if (vaultSharesAfterMint > shareLimit) revert ShareLimitExceeded(_vault, shareLimit);
+
+        // update the locked amount in advance
+        socket.sharesMinted = uint96(vaultSharesAfterMint);
+        uint256 reserveRatioBP = socket.reserveRatioBP;
+
+        uint256 totalEtherLocked = (STETH.getPooledEthByShares(vaultSharesAfterMint) * TOTAL_BASIS_POINTS) /
+            (TOTAL_BASIS_POINTS - reserveRatioBP);
+
+        if (totalEtherLocked > IStakingVault(_vault).locked()) {
+            IStakingVault(_vault).lock(totalEtherLocked);
+        }
+
+        // mint the shares to the recipient
+        STETH.mintExternalShares(_recipient, _amountOfShares);
+
+        // call the recipient with the provided data
+        (bool success, ) = _recipient.call(_data);
```

Comment:
Passing calldata as an arg feels like a critical, tbh. 

I think that's better to stick to some interface to limit potential impact surface. E.g. ERC-3156 have pretty decent interfaces we can partly reuse.